src/wallet.rs

```diff
@@ -1,82 +1,73 @@
-use crate::bitcoind_rpc::{AddressInfo, Client, Result, Unspent, WalletInfoResponse};
 use bitcoin::{Address, Amount, Transaction, Txid};
+use bitcoincore_rpc::bitcoincore_rpc_json::{
+    AddressType, GetAddressInfoResult, ListUnspentResultEntry, LoadWalletResult,
+};
+use bitcoincore_rpc::RawTx;
+use bitcoincore_rpc::RpcApi;
+use std::result;
 use url::Url;
 
 /// A wrapper to bitcoind wallet
 #[derive(Debug)]
 pub struct Wallet {
     name: String,
-    bitcoind_client: Client,
+    pub bitcoind_client: bitcoincore_rpc::Client,
 }
 
+pub type Result<T> = result::Result<T, bitcoincore_rpc::Error>;
+
 impl Wallet {
     /// Create a wallet on the bitcoind instance or use the wallet with the same name
     /// if it exists.
-    pub async fn new(name: &str, url: Url) -> Result<Self> {
-        let bitcoind_client = Client::new(url);
+    pub fn new(name: &str, url: &Url, auth: &bitcoincore_rpc::Auth) -> Result<Self> {
```

Comment:
:upside_down_face: I'm slightly concerned that `url` and `auth` are not orthogonal. Since they both come from the same source (a `Bitcoind` instance), we could create a method `new_wallet(&self, name: &str)` on `Bitcoind` which should be safer and easier to use.

Reply:
Makes sense. Checkout latest commit.

Reply:
Nice!